### ISSUE TITLE
Migrate from V4

### ISSUE THREAD

**Issue by KvL159** [YunoHost-Apps/invoiceninja5_ynh#1]
Thank you very much again for working on the V5 !

I tried the migration in V4 and I noticed that the links are not redirected properly.
My invoiceninja is installed in a subfolder : domain.tld/invoiceninja

When I click on "Migrate", it redirects to Yunohost SSO.

I opened the URL manually : domain.tld/invoiceninja/migration/start

It then shows the migrate screen, however when I click on "Continue", it redirects to Yunohost SSO again.

It may have something to do with ngninx config not redirecting ?

**Comment by rndmh3ro** [comment]
Hi @KvL159,

can you please update v4 and v5 to the latest releases? This was a bug in invoiceninja.
You probably have to use the github releases for now.

**Comment by KvL159** [comment]
Hi @rndmh3ro 

I have updated V4 to the lastest, it seems the migrationg page is redirected properly.

So I need to install V5 alongside V4 ? I haven't installed V5 yet

**Comment by KvL159** [comment]
OK, it needs to be installed alongside indeed!

I tried to install the ynh v5 package, I get an error during the install:
Package php7.4 is not available, but is referred to by another package. (and all the php7.4 packages)

So I need to enable Sury repo ?

**Comment by rndmh3ro** [comment]
> OK, it needs to be installed alongside indeed!

Yes, that's what the developers of invoiceninja want and suggest. 

> So I need to enable Sury repo ? 

Probably, yes. On my machine it was already done. I need to add this to the installation somehow.

**Comment by KvL159** [comment]
Hi @rndmh3ro 

I have enabled sury repo and installed the missing packages. The install succeeds, however when I open the page I get an error "403 Forbidden".
V5 is installed in a dedicated subdomain in the root / folder

I tried unistalling and reinstallating, no luck.

Any suggestions ?

There are no errors in the install log

**Comment by rndmh3ro** [comment]
Open /index.php. This should work. I need to modify the nginx config to make this work by default.

**Comment by KvL159** [comment]
Hi @rndmh3ro 

Ok that works, it opens the install wizard.

Regarding the DB/User step, is it created by your YNH package ? If yes, where to find the credentials.

Thank you !

**Comment by rndmh3ro** [comment]
Username and dB name is the name of the application, by default "invoiceninja5".
You can find the password in `/etc/yunohost/apps/invoiceninja5/settings.yml` or similar. It should be the variable `mysqlpwd`.

**Comment by rndmh3ro** [comment]
For the mail settings, use `log`.
The user you create should be the same as on your old invoiceninja 4 instance.

**Comment by KvL159** [comment]
Thank you for your feedback, I haven't have time to go further yet, I'll try ASAP.

I noticed that upstream has pushed a fix yesterday for the migration tool (V4):
https://github.com/invoiceninja/invoiceninja/releases/tag/v4.5.25

**Comment by rndmh3ro** [comment]
I tried this yesterday. Migration still not working. I'll have to open another issue with them.

**Comment by KvL159** [comment]
Hi @rndmh3ro !

I wanted to let you know that there seems to have been a couple a fixes in V4 for the migration tool:
https://github.com/invoiceninja/invoiceninja/releases/tag/v4.5.29

**Comment by rndmh3ro** [comment]
Hi @KvL159,

I'm also monitoring the new releases for v4 and v5. However for me, I still cannot migrate.

I updated the testing-branch to 4.5.30: https://github.com/YunoHost-Apps/invoiceninja_ynh
Maybe you can test, if you can migrate with this version?

**Comment by KvL159** [comment]
Hi @rndmh3ro 

I completed the setup of V5 and tried the migration, however I get the following error :
`Error: Make sure you did proper setup with V5 of Invoice Ninja, before starting migration.`

**Comment by rndmh3ro** [comment]
>  `Error: Make sure you did proper setup with V5 of Invoice Ninja, before starting migration.`

Yes, that's exactly the problem I face, too.

**Comment by KvL159** [comment]
Hello @rndmh3ro !

It seems that V5 has now a stable release, have you tried with the latest versions if the migration tool is now working properly ?

Thank you!

**Comment by rndmh3ro** [comment]
Hi @KvL159,

some updates:

* Migration works basically. This was not a problem with invoiceninja but with my nginx-configuration. Though the config still does not work correctly. I have a local workaround that doesn't help with a normal installation.
* I still need to completely automate the installation. However there's a problem with running curl to create a initial user.
* Upgrading/removing/backup doesn't work either.

**Comment by rndmh3ro** [comment]
With the latest changes installation of v5 and migration from v4 to v5 works. If there are any problems, please open a new issue.
